pkg/cmd/roachprod/grafana/annotations.go

```diff
@@ -42,25 +39,15 @@ func newGrafanaClient(
 		scheme = "https"
 
 		// Read in the service account key and audience, so we can retrieve the identity token.
-		grafanaKey := os.Getenv(ServiceAccountJson)
-		if grafanaKey == "" {
-			return nil, errors.Newf("%s env variable was not found", ServiceAccountJson)
-		}
-		grafanaAudience := os.Getenv(ServiceAccountAudience)
-		if grafanaAudience == "" {
-			return nil, errors.Newf("%s env variable was not found", ServiceAccountAudience)
+		if _, err := promhelperclient.SetPromHelperCredsEnv(ctx, false); err != nil {
```

Comment:
Documentation for `newGrafanaClient` could be updated to mention the function that reads credentials from env (maybe `SetPromHelperCredsEnv`), as `ServiceAccountJson` no longer exists.

Reply:
Also, the code reuse is good, but the fact that Grafana authentication works by calling `promhelperclient.SetPromHelperCredsEnv` is confusing because it suggests `promhelperclient` is involved.